cdap-app-fabric/src/main/java/co/cask/cdap/metadata/MetadataService.java

```diff
@@ -107,4 +106,11 @@ protected void shutDown() throws Exception {
     }
     LOG.info("Metadata HTTP service stopped");
   }
+
+  /**
+   * Adding this convenience method for debugging. Returns true if metadata migration is in progress.
+   */
+  public boolean isMigrationInProcess() {
+    return metadataMigrator.isRunning();
```

Comment:
Won't this be always running when CDAP is running? IT does not give an indication of the upgrade state if that's the case.

Reply:
The migrator will stop itself when there is no un-upgraded metadata tables - https://github.com/caskdata/cdap/blob/release/5.0/cdap-app-fabric/src/main/java/co/cask/cdap/metadata/MetadataMigrator.java#L95